### PR TITLE
Make ControlScript render step function faster

### PR DESCRIPTION
**Pull request by DarraghGriffin**
- Better manage the CachedHumanoid so we don't have to make sure it's in
the character when GetHumanoid is called
- Re-order if statement so humanoid.PlatformStand is not checked unless
isJumping is true.
- Cache VREnabled value so it doesn't need to be read every frame.
- Also removed some unused variables and unused code.

### CODE REVIEW

File: PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/ControlScript/MasterControl.rbxmx [hunk 1]
```diff
@@ -12,95 +12,115 @@
...
+	end
+
+	if characterChildRemovedConn then
+		characterChildRemovedConn:disconnect()
+	end	
+	
+	LocalCharacter = character
+	CachedHumanoid = LocalCharacter:FindFirstChildOfClass("Humanoid")
+	characterAncestryChangedConn = character.AncestryChanged:connect(function()
+		if not character:IsDescendantOf(workspace) then
```
Comment (hockeychaos) [inline PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/ControlScript/MasterControl.rbxmx]: when does this happen? do we need to worry about this case? should this really be, if the character has no parent, lets stop tracking it?

File: PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/ControlScript/MasterControl.rbxmx [hunk 2]
```diff
@@ -12,95 +12,115 @@
...
 			end
-			
-			moveFunc(LocalPlayer, adjustedMoveValue, true)	
-			
-			
 		end
+		
+		local adjustedMoveValue = moveValue
+		if VREnabled and workspace.CurrentCamera.HeadLocked then
+			local vrFrame = VRService:GetUserCFrame()
```
Comment (hockeychaos) [inline PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/ControlScript/MasterControl.rbxmx]: should we cache this function lookup?

Comment (hockeychaos) [inline PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/ControlScript/MasterControl.rbxmx]: also is this equivalent to UserInputService.UserHeadCFrame?

File: PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/ControlScript/MasterControl.rbxmx (hunk 1 above)
Comment (DarraghGriffin) [inline PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/ControlScript/MasterControl.rbxmx]: It only happens if the character is re-parented to be a descendant of another top level container by the developer. This case probably doesn't matter, we can just check character.Parent == nil here.

File: PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/ControlScript/MasterControl.rbxmx (hunk 2 above)
Comment (DarraghGriffin) [inline PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/ControlScript/MasterControl.rbxmx]: UserHeadCFrame is deprecated in favor of this method now, I'll cache the lookup though

**Pull request merged**